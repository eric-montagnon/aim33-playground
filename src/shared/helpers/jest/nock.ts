import { env } from "@src/env";
import nock from "nock";

export const nockWithEndpoint = (options?: nock.Options) =>
  nock(env.API_ENDPOINT, options);
