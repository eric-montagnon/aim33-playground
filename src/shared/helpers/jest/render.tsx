import { ThemeProvider } from "@emotion/react";
import { queryClient } from "@src/shared/infra/clients/queryClient";

import { QueryBoundaries } from "@src/shared/view/components/QueryBoundaries/QueryBoundaries.component";
import { ToastService } from "@src/shared/view/services/ToastService/Toast.service";
import { theme } from "@src/shared/view/theme/theme";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import {
  RenderOptions,
  // eslint-disable-next-line testing-library/no-manual-cleanup
  cleanup,
  render,
  screen,
  waitFor,
} from "@testing-library/react-native";
import React from "react";
import { SafeAreaProvider } from "react-native-safe-area-context";

// This may seem like a hack but it's reliable. afterEach hooks always run when we expect them too
let afterEachHook: (() => void) | undefined = undefined;
afterEach(() => {
  afterEachHook?.();
  afterEachHook = undefined;
});

export const wrapAndRender = (
  component: React.ReactElement,
  options?: RenderOptions,
) => {
  const frame = { x: 0, y: 0, width: 0, height: 0 };
  const insets = { top: 0, left: 0, right: 0, bottom: 0 };

  // In case there are multiple `renderWithQueries` calls in the same test, cleanup
  afterEachHook?.();

  // TODO: clean reusable solution logging
  // The idea is that in tests, you don't want to log known errors (you often write tests for these so it would be annoying)
  // But you want to log unknown errors, otherwise debugging tests is hell
  // setLogger(/* */);

  const waitForQueries = async () => {
    await waitFor(() => {
      assertNoPendingQueries(queryClient);
    });
  };

  afterEachHook = () => {
    // React cleanup should happen before waiting for queries, otherwise strange things happen
    cleanup();

    // Pending queries when a test end can cause multiple problems
    // - Not testing what you think you do (loaders in snapshots...)
    // - Test independence problems
    // - "Jest did not exit..." errors
    // -> This is important
    assertNoPendingQueries(queryClient);

    // This is a workaround for https://github.com/tannerlinsley/react-query/issues/1847
    queryClient.clear();
  };

  const renderResult = render(component, {
    wrapper: ({ children }) => (
      <QueryClientProvider client={queryClient}>
        <SafeAreaProvider
          initialMetrics={{
            frame,
            insets,
          }}
        >
          <ThemeProvider theme={theme}>
            <QueryBoundaries>{children}</QueryBoundaries>
            <ToastService.Toast />
          </ThemeProvider>
        </SafeAreaProvider>
      </QueryClientProvider>
    ),
    ...options,
  });

  return {
    ...renderResult,
    /**
     * When it's too complicated / not important for a test to wait for pending queries by checking UI elements
     * This function can be used as a fallback
     */
    waitForQueries,
  };
};

afterEach(() => {
  queryClient.clear();
});

export const waitForLoaders = () => {
  return waitFor(() => {
    const loader = screen.queryByTestId("activity-indicator");
    if (loader) throw new Error("Loader still visible");
  });
};

const assertNoPendingQueries = (queryClient: QueryClient) => {
  // Not using queryClient.isFetching() / isMutating() because we want to log details on the queries
  // const pendingQueries = queryClient
  //   .getQueryCache()
  //   .findAll(undefined)
  //   // The `filter` param of `.findAll` seems to be unreliable, so we filter manually (v3.16.0 de react-query, might have changed that)
  //   .filter((query) => query.isFetching())
  //   .map((query) => JSON.stringify(query.queryKey));
  const numberOfPendingQueries = queryClient.isFetching();

  if (numberOfPendingQueries > 0) {
    throw new Error(`Some queries still pending:\n${numberOfPendingQueries}`);
  }

  const pendingMutations = queryClient
    .getMutationCache()
    .findAll({})
    // The `filter` param of `.findAll` seems to be unreliable, so we filter manually (v3.16.0 de react-query, might have changed that)
    .filter((mutation) => mutation.state.status === "loading");

  if (pendingMutations.length > 0) {
    throw new Error(`${pendingMutations.length} mutations still pending`);
  }
};
