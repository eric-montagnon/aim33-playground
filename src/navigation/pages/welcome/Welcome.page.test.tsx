import { fireEvent, screen, waitFor } from "@testing-library/react-native";
import * as React from "react";

import { WelcomeScreenProps } from "@src/navigation/Navigator.interface";
import { WelcomePage } from "@src/navigation/pages/welcome/Welcome.page";
import { wrapAndRender } from "@src/shared/helpers/jest/render";

const navigateMock = jest.fn();
const welcomeTestProps = {
  navigation: {
    navigate: navigateMock,
  },
} as unknown as WelcomeScreenProps;

describe("Welcome Page", () => {
  it("displays properly", () => {
    wrapAndRender(<WelcomePage {...welcomeTestProps} />);

    expect(screen.getByText("Welcome to aim33 playground")).toBeTruthy();
    expect(screen).toMatchComponentSnapshot();
  });
  it("navigates to search when clicking on Dive In", async () => {
    wrapAndRender(<WelcomePage {...welcomeTestProps} />);

    fireEvent.press(screen.getByText("Dive In"));

    await waitFor(() => expect(navigateMock).toHaveBeenCalledWith("Search"));
  });
});
