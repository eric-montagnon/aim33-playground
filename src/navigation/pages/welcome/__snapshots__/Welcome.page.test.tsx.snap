// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Page displays properly 1`] = `
" Welcome to aim33 playground
Here you will be able to comfortably try our tools with us in the dojos.
 

Before we start, here are the components available :

Buttons :
Button.Primary
Button.Secondary
Button.Tertiary
IconButton :
Input :
Loader :
Switch :
Tags :
default Tag
valid Tag
error Tag
disabled Tag
We are now ready to start, let's dive in.
 

Dive In
"
`;

exports[`Welcome Page displays properly 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "height": 0,
        }
      }
    />
    <RCTScrollView
      alwaysBounceHorizontal={false}
      alwaysBounceVertical={false}
      contentContainerStyle={
        {
          "flexGrow": 1,
        }
      }
      keyboardShouldPersistTaps="handled"
      showsVerticalScrollIndicator={false}
      style={{}}
    >
      <View>
        <View
          style={
            {
              "padding": 24,
            }
          }
        >
          <Text
            numberOfLines={0}
            style={
              {
                "color": "#000000",
                "fontFamily": "OpenSans-Bold",
                "fontSize": 20,
                "lineHeight": 30,
                "textDecorationLine": "none",
              }
            }
          >
             Welcome to aim33 playground
          </Text>
          <View
            style={
              {
                "height": 20,
              }
            }
          />
          <View
            style={
              {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 16,
                "flex": 1,
                "padding": 16,
                "shadowColor": "#000000",
                "shadowOffset": {
                  "height": 3,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 2,
              }
            }
          >
            <Text
              numberOfLines={0}
              style={
                {
                  "color": "#000000",
                  "fontFamily": "OpenSans-Regular",
                  "fontSize": 16,
                  "lineHeight": 24,
                  "textDecorationLine": "none",
                }
              }
            >
              Here you will be able to comfortably try our tools with us in the dojos.
               
            </Text>
            <View
              style={
                {
                  "height": 10,
                }
              }
            />
            <Text
              numberOfLines={0}
              style={
                {
                  "color": "#000000",
                  "fontFamily": "OpenSans-Regular",
                  "fontSize": 16,
                  "lineHeight": 24,
                  "textDecorationLine": "none",
                }
              }
            >
              Before we start, here are the components available :
            </Text>
          </View>
          <View
            style={
              {
                "height": 25,
              }
            }
          />
          <Text
            numberOfLines={0}
            style={
              {
                "color": "#000000",
                "fontFamily": "OpenSans-Regular",
                "fontSize": 16,
                "lineHeight": 24,
                "textDecorationLine": "none",
              }
            }
          >
            Buttons :
          </Text>
          <View
            style={
              {
                "height": 10,
              }
            }
          />
          <View
            accessibilityLabel="Button.Primary"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#574BAA",
                "borderColor": "transparent",
                "borderRadius": 40,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingHorizontal": 16,
                "paddingVertical": 8,
                "shadowColor": "black",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 2,
              }
            }
          >
            <View
              isVisible={true}
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "opacity": 1,
                }
              }
            >
              <Text
                numberOfLines={0}
                style={
                  {
                    "color": "#FFFFFF",
                    "fontFamily": "OpenSans-Bold",
                    "fontSize": 20,
                    "lineHeight": 30,
                    "textDecorationLine": "none",
                  }
                }
              >
                Button.Primary
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "height": 10,
              }
            }
          />
          <View
            accessibilityLabel="Button.Secondary"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "#574BAA",
                "borderRadius": 40,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingHorizontal": 16,
                "paddingVertical": 8,
                "shadowColor": "black",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 2,
              }
            }
          >
            <View
              isVisible={true}
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "opacity": 1,
                }
              }
            >
              <Text
                numberOfLines={0}
                style={
                  {
                    "color": "#574BAA",
                    "fontFamily": "OpenSans-Bold",
                    "fontSize": 20,
                    "lineHeight": 30,
                    "textDecorationLine": "none",
                  }
                }
              >
                Button.Secondary
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "height": 10,
              }
            }
          />
          <View
            accessibilityLabel="Button.Tertiary"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderColor": "transparent",
                "borderRadius": 40,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingHorizontal": 16,
                "paddingVertical": 8,
                "shadowColor": "black",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 2,
              }
            }
          >
            <View
              isVisible={true}
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "opacity": 1,
                }
              }
            >
              <Text
                numberOfLines={0}
                style={
                  {
                    "color": "#574BAA",
                    "fontFamily": "OpenSans-Bold",
                    "fontSize": 20,
                    "lineHeight": 30,
                    "textDecorationLine": "underline",
                  }
                }
              >
                Button.Tertiary
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "height": 20,
              }
            }
          />
          <Text
            numberOfLines={0}
            style={
              {
                "color": "#000000",
                "fontFamily": "OpenSans-Regular",
                "fontSize": 16,
                "lineHeight": 24,
                "textDecorationLine": "none",
              }
            }
          >
            IconButton :
          </Text>
          <View
            accessibilityLabel="IconButton"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            style={
              {
                "backgroundColor": "#FFFFFF",
                "flexDirection": "row",
                "justifyContent": "center",
                "opacity": 1,
                "paddingHorizontal": 16,
                "paddingVertical": 8,
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight="24"
              bbWidth="24"
              fill="none"
              focusable={false}
              height={24}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                  "flex": 0,
                  "height": 24,
                  "width": 24,
                }
              }
              testID="error-icon"
              vbHeight={24}
              vbWidth={24}
              width={24}
            >
              <RNSVGGroup
                fill={null}
                propList={
                  [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="M12.0005 15C11.8027 15 11.6093 15.0587 11.4449 15.1686C11.2804 15.2784 11.1523 15.4346 11.0766 15.6173C11.0009 15.8001 10.9811 16.0011 11.0197 16.1951C11.0583 16.3891 11.1535 16.5673 11.2933 16.7071C11.4332 16.847 11.6114 16.9422 11.8054 16.9808C11.9993 17.0194 12.2004 16.9996 12.3831 16.9239C12.5659 16.8482 12.722 16.72 12.8319 16.5556C12.9418 16.3911 13.0005 16.1978 13.0005 16C13.0005 15.7348 12.8951 15.4805 12.7076 15.2929C12.52 15.1054 12.2657 15 12.0005 15ZM22.6705 16.47L14.6205 2.47003C14.3603 2.00354 13.9802 1.61498 13.5196 1.3445C13.0591 1.07401 12.5346 0.931396 12.0005 0.931396C11.4663 0.931396 10.9419 1.07401 10.4813 1.3445C10.0207 1.61498 9.64065 2.00354 9.38046 2.47003L1.38046 16.47C1.11125 16.924 0.966598 17.441 0.9611 17.9688C0.955602 18.4966 1.08945 19.0166 1.34914 19.4761C1.60883 19.9356 1.98516 20.3185 2.44014 20.5861C2.89512 20.8536 3.41264 20.9964 3.94046 21H20.0605C20.5925 21.0053 21.1164 20.8689 21.5784 20.6049C22.0403 20.341 22.4238 19.9589 22.6894 19.4978C22.9551 19.0368 23.0933 18.5134 23.09 17.9814C23.0866 17.4493 22.9418 16.9277 22.6705 16.47ZM20.9405 18.47C20.8528 18.626 20.7249 18.7556 20.5701 18.8453C20.4154 18.935 20.2393 18.9815 20.0605 18.98H3.94046C3.76157 18.9815 3.58556 18.935 3.43077 18.8453C3.27599 18.7556 3.14811 18.626 3.06046 18.47C2.97269 18.318 2.92648 18.1456 2.92648 17.97C2.92648 17.7945 2.97269 17.622 3.06046 17.47L11.0605 3.47003C11.1444 3.30623 11.2719 3.16876 11.4289 3.07277C11.5859 2.97678 11.7664 2.92599 11.9505 2.92599C12.1345 2.92599 12.315 2.97678 12.472 3.07277C12.629 3.16876 12.7565 3.30623 12.8405 3.47003L20.8905 17.47C20.9897 17.6199 21.0467 17.7937 21.0555 17.9732C21.0643 18.1527 21.0245 18.3312 20.9405 18.49V18.47ZM12.0005 7.00003C11.7352 7.00003 11.4809 7.10538 11.2933 7.29292C11.1058 7.48046 11.0005 7.73481 11.0005 8.00003V12C11.0005 12.2652 11.1058 12.5196 11.2933 12.7071C11.4809 12.8947 11.7352 13 12.0005 13C12.2657 13 12.52 12.8947 12.7076 12.7071C12.8951 12.5196 13.0005 12.2652 13.0005 12V8.00003C13.0005 7.73481 12.8951 7.48046 12.7076 7.29292C12.52 7.10538 12.2657 7.00003 12.0005 7.00003Z"
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "fill",
                    ]
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <View
            style={
              {
                "height": 20,
              }
            }
          />
          <Text
            numberOfLines={0}
            style={
              {
                "color": "#000000",
                "fontFamily": "OpenSans-Regular",
                "fontSize": 16,
                "lineHeight": 24,
                "textDecorationLine": "none",
              }
            }
          >
            Input :
          </Text>
          <View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              isDisabled={false}
              isError={false}
              isFocused={false}
              style={
                {
                  "backgroundColor": "#FFFFFF",
                  "borderColor": "#E5E5E5",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "height": 40,
                  "justifyContent": "space-between",
                  "paddingHorizontal": 16,
                  "paddingVertical": 8,
                }
              }
            >
              <TextInput
                editable={true}
                style={
                  {
                    "color": "#000000",
                    "flex": 1,
                    "fontFamily": "OpenSans-Regular",
                    "fontSize": 20,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              {
                "height": 20,
              }
            }
          />
          <Text
            numberOfLines={0}
            style={
              {
                "color": "#000000",
                "fontFamily": "OpenSans-Regular",
                "fontSize": 16,
                "lineHeight": 24,
                "textDecorationLine": "none",
              }
            }
          >
            Loader :
          </Text>
          <ActivityIndicator />
          <View
            style={
              {
                "height": 20,
              }
            }
          />
          <Text
            numberOfLines={0}
            style={
              {
                "color": "#000000",
                "fontFamily": "OpenSans-Regular",
                "fontSize": 16,
                "lineHeight": 24,
                "textDecorationLine": "none",
              }
            }
          >
            Switch :
          </Text>
          <View
            accessibilityHint="Switch"
            animatedStyle={
              {
                "value": {
                  "backgroundColor": "rgba(204, 204, 204, 1)",
                  "borderColor": "rgba(0, 0, 0, 0)",
                  "borderWidth": 1,
                },
              }
            }
            collapsable={false}
            style={
              {
                "backgroundColor": "rgba(204, 204, 204, 1)",
                "borderColor": "rgba(0, 0, 0, 0)",
                "borderRadius": 20.5,
                "borderWidth": 1,
                "height": 41,
                "justifyContent": "center",
                "width": 74,
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": false,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              animatedStyle={
                {
                  "value": {},
                }
              }
              collapsable={false}
              focusable={true}
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row-reverse",
                  "height": "100%",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                style={
                  {
                    "width": 10,
                  }
                }
              />
              <View
                style={
                  {
                    "flex": 1,
                  }
                }
              />
              <View
                animatedStyle={
                  {
                    "value": {
                      "backgroundColor": "rgba(255, 255, 255, 1)",
                      "borderColor": "rgba(0, 0, 0, 0)",
                      "borderWidth": 1,
                    },
                  }
                }
                collapsable={false}
                layout={
                  LinearTransition {
                    "build": [Function],
                    "callbackV": undefined,
                    "dampingRatioV": undefined,
                    "dampingV": undefined,
                    "delayV": undefined,
                    "durationV": 300,
                    "easingV": undefined,
                    "initialValues": undefined,
                    "massV": undefined,
                    "overshootClampingV": undefined,
                    "randomizeDelay": false,
                    "reduceMotionV": "system",
                    "restDisplacementThresholdV": undefined,
                    "restSpeedThresholdV": undefined,
                    "rotateV": undefined,
                    "stiffnessV": undefined,
                    "type": undefined,
                  }
                }
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "rgba(255, 255, 255, 1)",
                    "borderColor": "rgba(0, 0, 0, 0)",
                    "borderRadius": 15.5,
                    "borderWidth": 1,
                    "height": 31,
                    "justifyContent": "center",
                    "width": 31,
                  }
                }
              />
              <View
                style={
                  {
                    "width": 5,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              {
                "height": 20,
              }
            }
          />
          <Text
            numberOfLines={0}
            style={
              {
                "color": "#000000",
                "fontFamily": "OpenSans-Regular",
                "fontSize": 16,
                "lineHeight": 24,
                "textDecorationLine": "none",
              }
            }
          >
            Tags :
          </Text>
          <View
            style={
              {
                "height": 10,
              }
            }
          />
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": true,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            style={
              {
                "alignItems": "center",
                "alignSelf": "flex-start",
                "backgroundColor": "#FFFFFF",
                "borderColor": "#574BAA",
                "borderRadius": 40,
                "borderWidth": 1,
                "flexDirection": "row",
                "opacity": 1,
                "padding": 8,
              }
            }
          >
            <Text
              numberOfLines={0}
              style={
                {
                  "color": "#574BAA",
                  "fontFamily": "OpenSans-Bold",
                  "fontSize": 12,
                  "lineHeight": 18,
                  "textDecorationLine": "none",
                }
              }
            >
              default Tag
            </Text>
          </View>
          <View
            style={
              {
                "height": 5,
              }
            }
          />
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": true,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            style={
              {
                "alignItems": "center",
                "alignSelf": "flex-start",
                "backgroundColor": "#FFFFFF",
                "borderColor": "#5CC689",
                "borderRadius": 40,
                "borderWidth": 1,
                "flexDirection": "row",
                "opacity": 1,
                "padding": 8,
              }
            }
          >
            <Text
              numberOfLines={0}
              style={
                {
                  "color": "#5CC689",
                  "fontFamily": "OpenSans-Bold",
                  "fontSize": 12,
                  "lineHeight": 18,
                  "textDecorationLine": "none",
                }
              }
            >
              valid Tag
            </Text>
          </View>
          <View
            style={
              {
                "height": 5,
              }
            }
          />
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": true,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            style={
              {
                "alignItems": "center",
                "alignSelf": "flex-start",
                "backgroundColor": "#FFFFFF",
                "borderColor": "#E9263D",
                "borderRadius": 40,
                "borderWidth": 1,
                "flexDirection": "row",
                "opacity": 1,
                "padding": 8,
              }
            }
          >
            <Text
              numberOfLines={0}
              style={
                {
                  "color": "#E9263D",
                  "fontFamily": "OpenSans-Bold",
                  "fontSize": 12,
                  "lineHeight": 18,
                  "textDecorationLine": "none",
                }
              }
            >
              error Tag
            </Text>
          </View>
          <View
            style={
              {
                "height": 5,
              }
            }
          />
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": true,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            style={
              {
                "alignItems": "center",
                "alignSelf": "flex-start",
                "backgroundColor": "#F2F2F2",
                "borderColor": "#CCCCCC",
                "borderRadius": 40,
                "borderWidth": 1,
                "flexDirection": "row",
                "opacity": 1,
                "padding": 8,
              }
            }
          >
            <Text
              numberOfLines={0}
              style={
                {
                  "color": "#CCCCCC",
                  "fontFamily": "OpenSans-Bold",
                  "fontSize": 12,
                  "lineHeight": 18,
                  "textDecorationLine": "none",
                }
              }
            >
              disabled Tag
            </Text>
          </View>
          <View
            style={
              {
                "height": 10,
              }
            }
          />
          <View
            style={
              {
                "height": 35,
              }
            }
          />
          <View
            style={
              {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 16,
                "flex": 1,
                "padding": 16,
                "shadowColor": "#000000",
                "shadowOffset": {
                  "height": 3,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 2,
              }
            }
          >
            <Text
              numberOfLines={0}
              style={
                {
                  "color": "#000000",
                  "fontFamily": "OpenSans-Regular",
                  "fontSize": 16,
                  "lineHeight": 24,
                  "textDecorationLine": "none",
                }
              }
            >
              We are now ready to start, let's dive in.
               
            </Text>
          </View>
          <View
            style={
              {
                "height": 20,
              }
            }
          />
          <View
            accessibilityLabel="Dive In"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#574BAA",
                "borderColor": "transparent",
                "borderRadius": 40,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingHorizontal": 16,
                "paddingVertical": 8,
                "shadowColor": "black",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 2,
              }
            }
          >
            <View
              isVisible={true}
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "opacity": 1,
                }
              }
            >
              <Text
                numberOfLines={0}
                style={
                  {
                    "color": "#FFFFFF",
                    "fontFamily": "OpenSans-Bold",
                    "fontSize": 20,
                    "lineHeight": 30,
                    "textDecorationLine": "none",
                  }
                }
              >
                Dive In
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "height": 0,
            }
          }
        />
      </View>
    </RCTScrollView>
  </View>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "transform": [
          {
            "translateY": -0,
          },
        ],
      }
    }
    testID="toastAnimatedContainer"
  >
    <View
      style={
        {
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#EBFBF1",
            "flex": 1,
            "flexDirection": "row",
            "padding": 16,
            "paddingTop": 16,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="24"
          bbWidth="24"
          fill="none"
          focusable={false}
          height={24}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "flex": 0,
              "height": 24,
              "width": 24,
            }
          }
          vbHeight={24}
          vbWidth={24}
          width={24}
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M18.7104 7.20998C18.6175 7.11625 18.5069 7.04186 18.385 6.99109C18.2632 6.94032 18.1324 6.91418 18.0004 6.91418C17.8684 6.91418 17.7377 6.94032 17.6159 6.99109C17.494 7.04186 17.3834 7.11625 17.2904 7.20998L9.84044 14.67L6.71044 11.53C6.61392 11.4367 6.49998 11.3634 6.37512 11.3142C6.25026 11.265 6.11694 11.2409 5.98276 11.2432C5.84858 11.2455 5.71617 11.2743 5.59309 11.3278C5.47001 11.3812 5.35868 11.4585 5.26544 11.555C5.1722 11.6515 5.09889 11.7654 5.04968 11.8903C5.00048 12.0152 4.97635 12.1485 4.97867 12.2827C4.98099 12.4168 5.00972 12.5493 5.06321 12.6723C5.1167 12.7954 5.19392 12.9067 5.29044 13L9.13044 16.84C9.2234 16.9337 9.334 17.0081 9.45586 17.0589C9.57772 17.1096 9.70843 17.1358 9.84044 17.1358C9.97245 17.1358 10.1032 17.1096 10.225 17.0589C10.3469 17.0081 10.4575 16.9337 10.5504 16.84L18.7104 8.67998C18.8119 8.58634 18.893 8.47269 18.9484 8.34619C19.0038 8.21969 19.0324 8.08308 19.0324 7.94498C19.0324 7.80688 19.0038 7.67028 18.9484 7.54378C18.893 7.41728 18.8119 7.30363 18.7104 7.20998V7.20998Z"
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <View
          style={
            {
              "flex": 1,
              "flexDirection": "column",
              "paddingLeft": 8,
            }
          }
        >
          <Text
            accessibilityRole="alert"
            numberOfLines={0}
            style={
              {
                "color": "#000000",
                "fontFamily": "OpenSans-Bold",
                "fontSize": 20,
                "lineHeight": 30,
                "textDecorationLine": "none",
              }
            }
          />
          <Text
            accessibilityRole="alert"
            numberOfLines={0}
            style={
              {
                "color": "#000000",
                "fontFamily": "OpenSans-Regular",
                "fontSize": 16,
                "lineHeight": 24,
                "textDecorationLine": "none",
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
